dev-packages/rollup-utils/plugins/make-esm-plugin.mjs

```diff
@@ -0,0 +1,16 @@
+/**
+ * Outputs a package.json file with {type: module} in the root of the output directory so that Node
+ * treats .js files as ESM.
+ */
+export function makePackageNodeEsm() {
+  return {
+    name: 'make-package-node-esm',
+    generateBundle() {
+      this.emitFile({
+        type: 'asset',
+        fileName: 'package.json',
+        source: '{ "type": "module" }',
```

Comment:
lol

are we sure we don't need name or any entrypoints for this package.json?

Reply:
The Node **Determining module system** docs [say](https://nodejs.org/api/packages.html#determining-module-system):
> Node.js will treat the following as [ES modules](https://nodejs.org/api/esm.html) when passed to `node` as the initial input, or when referenced by `import` statements or `import()` expressions:
> - Files with a `.js` extension when the nearest parent `package.json` file contains a top-level ["type"](https://nodejs.org/api/packages.html#type) field with a value of "module".

I checked and It does seem like this is a common approach:
https://github.com/search?q=echo+%27%7B%5C%22type%5C%22%3A+%5C%22module%5C%22%7D&type=code

Reply:
I actually did this 5 months ago for the Electron SDK and forgot since then 😂
https://github.com/getsentry/sentry-electron/blob/02322af325a564f01fc58a8c07bc89d1fda8a69e/rollup.config.js#L20-L30